### PR TITLE
Add support for rendering doubles using Grisu 2

### PR DESCRIPTION
**Pull request by jystic**
Moving this from icicle to here so we can call it from Haskell.

Also, we finally have a roundtrip test for the double parsing / rendering!

! @amosr @tranma 
/jury approved @amosr

### CODE REVIEW

File: src/Anemone/Embed.hs [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
+  $(FileEmbed.embedFile "csrc/anemone_base.h")
 
-anemone_mempool_h :: ByteString
-anemone_mempool_h = $(FileEmbed.embedFile "csrc/anemone_mempool.h")
+anemone_grisu2_h :: ByteString
+anemone_grisu2_h =
+  $(FileEmbed.embedFile "csrc/anemone_grisu2.h")
 
-anemone_mempool_c :: ByteString
-anemone_mempool_c = $(FileEmbed.embedFile "csrc/anemone_mempool.c")
```
Comment (jystic) [inline src/Anemone/Embed.hs]: I figured it didn't make sense to expose the implementation, `anemone_mempool.c`, we might as well link to it using `rdynamic` like we do with `zebra` and `piano`.

File: test/Test/Anemone/Roundtrip.hs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import           System.IO (IO)
+
+
+prop_roundtrip_double :: Property
+prop_roundtrip_double =
+  gamble arbitrary $ \(n :: Int64) ->
+  gamble arbitrary $ \(x :: Double) ->
+  let
+    original =
+      0 + x * fromIntegral n
```
Comment (jystic) [inline test/Test/Anemone/Roundtrip.hs]: This `0 +` is not strictly necessary, but it means we get more interesting counterexamples if the code below is changed to `diff_i64 == 0` (otherwise we just get the `0.0` vs `-0.0` case)

tl;dr adding zero to a floating point number normalises -0 => 0

**Comment by amosr** [comment]
👍
